### ISSUE TITLE
Incorrect binding of suppressed target-typed array initializer

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#32674]
```C#
        [Fact]
        public void SuppressNullableWarning_TargetTypedArrayInitializer()
        {
            var comp = CreateCompilation(@"
class C
{
    void M()
    {
        string[] s = (new[] { null })!;
    }
}");
            comp.VerifyDiagnostics(
                // (6,22): error CS0029: Cannot implicitly convert type '?[]' to 'string[]'
                //         string[] s = (new[] { null })!;
                Diagnostic(ErrorCode.ERR_NoImplicitConv, "(new[] { null })!").WithArguments("?[]", "string[]").WithLocation(6, 22),
                // (6,23): error CS0826: No best type found for implicitly-typed array
                //         string[] s = (new[] { null })!;
                Diagnostic(ErrorCode.ERR_ImplicitlyTypedArrayNoBestType, "new[] { null }").WithLocation(6, 23)
                );
        }
```

**Comment by gafter** [comment]
[There is no such thing as a target-typed array initializer](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABADAAmwEYBuAWAChsBmQgJnwGF8BvS/Dw27AFnwFkAFAEpWlAJDjiuANoBdfFHwBefADs4Ad3mt1AVwA2B/AF9yFTqcomgA===).  This program should give the same errors without the `!`.
